### PR TITLE
Filter task run

### PR DESCRIPTION
**Pull request by abhinandan13jan**
## Fixes 

https://issues.redhat.com/browse/HACBS-1170?

## Description

Adds filter for Task name to filter by metadata.name and spec.taskRef.name

## Type of change

- [x] Feature
- [ ] Bugfix
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Screen shots / Gifs for design review 

Screenshot 2022-09-28 at 6 20 22 PM
Screenshot 2022-09-28 at 6 20 34 PM
Screenshot 2022-09-28 at 6 20 58 PM

## Browser conformance: 

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: src/hacbs/components/TaskRunListView/TaskRunListView.tsx [hunk 1]
```diff
@@ -20,6 +36,44 @@ const TaskRunListView: React.FC<Props> = ({ pipelineName, namespace }) => {
     },
   });
 
+  const [nameFilter, setNameFilter] = React.useState<string>('');
```
Comment (christianvogt) [inline src/hacbs/components/TaskRunListView/TaskRunListView.tsx:39]: Let's update the query param to be consistent with other pages.
```suggestion
 const [nameFilter, setNameFilter] = useSearchParam('name', '');
```

Use The `ComponentListView` as an example.

File: src/hacbs/components/TaskRunListView/TaskRunListView.tsx [hunk 2]
```diff
@@ -20,6 +36,44 @@ const TaskRunListView: React.FC<Props> = ({ pipelineName, namespace }) => {
...
+
+  taskRuns?.sort(
+    (a, b) => +new Date(a.metadata.creationTimestamp) - +new Date(b.metadata.creationTimestamp),
+  );
```
Comment (christianvogt) [inline src/hacbs/components/TaskRunListView/TaskRunListView.tsx:43]: Now that this page could re-render due to filtering, capture the sorted tasks inside a memo.

Comment (abhinandan13jan) [reply src/hacbs/components/TaskRunListView/TaskRunListView.tsx]: updated

**Comment by abhinandan13jan** [comment]
Screenshot 2022-09-29 at 6 26 45 PM

**Comment by christianvogt** [comment]
@abhinandan13jan tests are failing

**Pull request merged**